Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
```

Comment:
To me this mostly looks at if this was data (and not functionality). Also, as far as I see, the types of questions is not an extension point for consumers of the service, i.e. the service comes with a fixed amount of question types that cannot be extended by some outsider. This to me hints at the fact, that we also might need a base-class for the question and the interface is just a way to shield consumers from the innarts of the service.

Reply:
yes, i guess you're indeed right as far as I understood you. a base class will come, as it also now exists. it will contain a lot of internaly used logic as it is also the state in the current implementation.